model/project_parser.go

```diff
@@ -1051,3 +1080,199 @@ func evaluateDependsOn(tse *tagSelectorEvaluator, tgse *tagSelectorEvaluator, vs
+	for _, task := range pp1.Tasks {
+		taskNameExist[task.Name] = true
+	}
+	for _, task := range pp2.Tasks {
+		if _, ok := taskNameExist[task.Name]; ok {
+			catcher.Add(errors.Errorf("task '%s' has been declared already", task.Name))
+		} else {
+			pp1.Tasks = append(pp1.Tasks, task)
+			taskNameExist[task.Name] = true
+		}
+	}
+
+	taskGroupNameExist := map[string]bool{}
+	for _, taskGroup := range pp1.TaskGroups {
+		taskGroupNameExist[taskGroup.Name] = true
+	}
+	for _, taskGroup := range pp2.TaskGroups {
+		if _, ok := taskGroupNameExist[taskGroup.Name]; ok {
+			catcher.Add(errors.Errorf("taskGroup '%s' has been declared already", taskGroup.Name))
+		} else {
+			pp1.TaskGroups = append(pp1.TaskGroups, taskGroup)
+			taskGroupNameExist[taskGroup.Name] = true
+		}
+	}
+
+	parameterKeyExist := map[string]bool{}
+	for _, parameter := range pp1.Parameters {
+		parameterKeyExist[parameter.Key] = true
+	}
+	for _, parameter := range pp2.Parameters {
+		if _, ok := parameterKeyExist[parameter.Key]; ok {
+			catcher.Add(errors.Errorf("parameter key '%s' has been declared already", parameter.Key))
+		} else {
+			pp1.Parameters = append(pp1.Parameters, parameter)
+			parameterKeyExist[parameter.Key] = true
+		}
+	}
+
+	moduleExist := map[string]bool{}
+	for _, module := range pp1.Modules {
+		moduleExist[module.Name] = true
+	}
+	for _, module := range pp2.Modules {
+		if _, ok := moduleExist[module.Name]; ok {
+			catcher.Add(errors.Errorf("module '%s' has been declared already", module.Name))
+		} else {
+			pp1.Modules = append(pp1.Modules, module)
+			moduleExist[module.Name] = true
+		}
+	}
+
+	for key, val := range pp2.Functions {
+		if _, ok := pp1.Functions[key]; ok {
+			catcher.Add(errors.Errorf("function '%s' has been declared already", key))
+		} else {
+			pp1.Functions[key] = val
+		}
+	}
+	// Unordered list, don't consider naming conflict
+	pp1.Ignore = mergeParserStringSlice(pp1.Ignore, pp2.Ignore)
+	// for _, log := range pp2.Loggers.Agent {
+	// 	pp1.Loggers.Agent = append(pp1.Loggers.Agent, log)
+	// }
+	// pp1.Loggers.Agent = append(pp1.Loggers.Agent, pp2.Loggers.Agent...)
+
+	// not sure why this doesn't work
+	// pp1.Loggers = mergeAllLogs(pp1.Loggers, pp2.Loggers)
+
+	// Ordered list, cannot be defined for more than one yaml
+	if pp1.Pre != nil && pp2.Pre != nil {
+		catcher.Add(errors.New("pre can only be defined in one yaml"))
+	} else {
+		if pp2.Pre != nil {
+			pp1.Pre = pp2.Pre
+		}
+	}
+
+	if pp1.Post != nil && pp2.Post != nil {
+		catcher.Add(errors.New("post can only be defined in one yaml"))
+	} else {
+		if pp2.Post != nil {
+			pp1.Post = pp2.Post
+		}
+	}
+
+	if pp1.Timeout != nil && pp2.Timeout != nil {
+		catcher.Add(errors.New("timeout can only be defined in one yaml"))
+	} else {
+		if pp2.Timeout != nil {
+			pp1.Timeout = pp2.Timeout
+		}
+	}
+
+	if pp1.EarlyTermination != nil && pp2.EarlyTermination != nil {
+		catcher.Add(errors.New("earlyTermination can only be defined in one yaml"))
```

Comment:
better to use what the thing is as in english opposed to the variable name itself in errors (early termination, batchtime, display name, etc)